Sources/Remote Logging/Crash Logging/CrashLogging.swift

```diff
@@ -146,6 +146,22 @@ public class CrashLogging {
 // MARK: - Manual Error Logging
 public extension CrashLogging {
 
+    /// Writes a JavaScript exception to the Crash Logging system, including its stack trace.
+    /// Note that this function is provided mainly for hybrid sources like React Native.
+    ///
+    /// - Parameters:
+    ///   - exception: The exception object
+    ///   - callback: Callback triggered upon completion
+    func logJavaScriptException(_ jsException: JSException, callback: @escaping () -> Void) {
```

Comment:
The `callback` argument _feels_ redundant to me. I have asked about removing it in https://github.com/WordPress/gutenberg/pull/59221#pullrequestreview-1907518762.

Reply:
I shared a comment about this [here](https://github.com/WordPress/gutenberg/pull/59221/files/c017e122aa58dbfcc5e33fad72a1b9207ac77880#r1507239536). In React Native, the communication between JavaScript and native is asynchronous so it's common to use callbacks.

Reply:
I'm wondering about the callback too. It doesn't seem to add much value when passed this far down the call stack. As you suggest in [this comment](https://github.com/WordPress/gutenberg/pull/59221#discussion_r1507248660), it's only needed to inform about a malformed `rawException`. Even there I'm not sure how useful it will be.

Reply:
@crazytonyli Regarding [this comment](https://github.com/WordPress/gutenberg/pull/59221#discussion_r1508336687), I think we need to have the callback here in Tracks because we need to guarantee that it's triggered after flushing all queued events:

https://github.com/Automattic/Automattic-Tracks-iOS/blob/07a7012a5af7a5e8c47bc5959e5d1331a8cffe6f/Sources/Remote%20Logging/Crash%20Logging/CrashLogging.swift#L159-L162

This implementation is based on other functions we have to ensure that events are sent, e.g.:
https://github.com/Automattic/Automattic-Tracks-iOS/blob/07a7012a5af7a5e8c47bc5959e5d1331a8cffe6f/Sources/Remote%20Logging/Crash%20Logging/CrashLogging.swift#L247-L274